### PR TITLE
Convert IntentLauncherAndroid to unimodule

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Making modules universal.

# How

- Moved `IntentLauncherAndroid` module to universal module `expo-intent-launcher`.
- Changed name to `IntentLauncher` - we no longer use `Android/iOS` suffixes, so I just took the opportunity and renamed it.
- Added more configuration options (mostly copied from https://github.com/poberwong/react-native-intent-launcher).
- Added `ModuleNotFoundException` and `CurrentActivityNotFoundException` to `expo-errors` as they are going to be used in multiple modules.

# Test Plan

`native-component-list` example works 👌

### CODE REVIEW

File: android/expoview/build.gradle [hunk 1]
```diff
@@ -220,6 +220,7 @@ dependencies {
   api project(':expo-image-loader-interface')
   api project(':expo-image-manipulator')
   api project(':expo-image-picker')
+  api project(':expo-intent-launcher')
```
Comment (sjchmiela) [inline android/expoview/build.gradle]: missing `compileOnly host.exp.exponent:expo-intent-launcher`?

File: packages/expo-intent-launcher/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# expo-intent-launcher
+
+`expo-intent-launcher` provides a way to launch Android intents. e.g. - opening a specific activities.
```
Comment (sjchmiela) [inline packages/expo-intent-launcher/README.md]:
```suggestion
`expo-intent-launcher` provides a way to launch Android intents. e.g. opening a specific activity.
```

File: packages/expo-intent-launcher/android/build.gradle [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  throw new GradleException(
+      '\'expo-core.gradle\' was not found in the usual Flutter or React Native dependency locations. ' +
+          'This package can only be used in such projects. Are you sure you\'ve installed the dependencies properly?')
+}
+
+dependencies {
+  expendency 'expo-core'
+  expendency 'expo-errors'
+
+  api 'com.android.support:support-annotations:28.0.0'
```
Comment (sjchmiela) [inline packages/expo-intent-launcher/android/build.gradle]: Is it ok to include support-annotations in v28 whereas in other expo modules/expoview we use 27.1.1 iirc?

File: packages/expo-intent-launcher/android/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<manifest package="expo.modules.template">
```
Comment (sjchmiela) [inline packages/expo-intent-launcher/android/src/main/AndroidManifest.xml]:
```suggestion
<manifest package="expo.modules.intentlauncher">
```

File: packages/expo-intent-launcher/android/src/main/java/expo/modules/intentlauncher/IntentLauncherModule.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+      intent.putExtras(params.getArguments(ATTR_EXTRA).toBundle());
+    }
+    if (params.containsKey(ATTR_FLAGS)) {
+      intent.addFlags(params.getInt(ATTR_FLAGS));
+    }
+    if (params.containsKey(ATTR_CATEGORY)) {
+      intent.addCategory(params.getString(ATTR_CATEGORY));
+    }
+
+    if (mUIManager != null) {
```
Comment (sjchmiela) [inline packages/expo-intent-launcher/android/src/main/java/expo/modules/intentlauncher/IntentLauncherModule.java]: You've already checked in line 67 that `mUIManager != null`

File: packages/expo-intent-launcher/src/IntentLauncher.ts [hunk 6]
```diff
@@ -101,15 +101,32 @@ export const ACTION_ZEN_MODE_SCHEDULE_RULE_SETTINGS =
...
+  extra?: object;
+}
+
+export enum ResultCode {
+  Success = -1,
+  Canceled = 0,
+  FirstUser = 1,
+}
+
+export async function startActivityAsync(params: IntentParams = {}): Promise<IntentResult> {
```
Comment (sjchmiela) [inline packages/expo-intent-launcher/src/IntentLauncher.ts]: I'm not sure I'm really a fan of putting all the args into a big object like this, I would rather leave `activity:string` as the first, required argument and then let developers pass extra `data, uri, mime` in an options object as the second argument, but that's only my opinion 🤷‍♂️

**Review by sjchmiela — COMMENTED** [review]
Some useful nitpicking

File: packages/expo-intent-launcher/src/IntentLauncher.ts (hunk 6 above)
Comment (bbarthec) [reply packages/expo-intent-launcher/src/IntentLauncher.ts]: @sjchmiela but I cannot see `activity` argument anymore. All these params seems to be optional now, right @tsapeta?

Comment (tsapeta) [reply packages/expo-intent-launcher/src/IntentLauncher.ts]: `activity` has been replaced with `action` and actually there is a mistake in the interface above, `action` argument is required.

Comment (bbarthec) [reply packages/expo-intent-launcher/src/IntentLauncher.ts]: Ah, so I would agree with @sjchmiela on having this argument as separate parameter 😄

Comment (bbarthec) [reply packages/expo-intent-launcher/src/IntentLauncher.ts]: Summon @dsokal 	😛

**Review by bbarthec — APPROVED** [review]

**Comment by kandarpdave** [comment]
@tsapeta Does this change allow us to open any intent in Android? For example, if I want to share an image from my app to Instagram, would passing `com.android.instagram` in `packageName` open IG? (That package string may not be correct, I'm just guessing here). Thanks!

**Comment by sjchmiela** [comment]
I think so!

**Pull request merged**